src/Views/HardwareView.vala

```diff
@@ -280,6 +282,19 @@ public class About.HardwareView : Gtk.Grid {
 
         return 0;
     }
+
+    private string get_storage_type () {
+        string storage = "";
+        var drive = new UDisks.DriveSkeleton ();
```

Comment:
As the class is named "DriveSkeleton" and it's description says `Creates a skeleton object for the D-Bus interface org.freedesktop.UDisks2.Drive`, it's not an actual implementation nor something that actually gets the information about the drive that contains the system partition. It doesn't know magically what drive to check, it's just a dummy object.

In this case, it's coincidental that everyone that tested this branch had an SSD. It shows SSD, because by default the `rotation_rate` is 0. This doesn't work currently with HDD's.

At this point you should try to use the [UDisks.Client](https://valadoc.org/udisks2/UDisks.Client.html) class and then try to enumerate on the drives, see which is the "/" and check the rotation_rate of that.